### ISSUE TITLE
Retire one or more error-handling modes?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#20856]
Chapel currently has three error-handling modes, documented here:  https://chapel-lang.org/docs/technotes/errorHandling.html#error-handling-modes  I haven't thought about the fact that there are these three modes in years, which makes me wonder whether we could jettison one or two of them.  I do continue to like that I can write file-scope code without worrying about catching every error, which might suggest supporting just two and deprecating the third.  But it'd be good to review what we have here, decide what we're committing to, move it into the spec, and retire this technical note.

Related:  The `prototype` keyword has always been a slightly odd one, and @mppf proposes that perhaps it should be an annotation.  I'm intrigued by that...

**Comment by lydia-duncan** [comment]
I'd forgotten there were three modes as well.  I would be a bit worried we'd get rid of a mode someone was relying heavily on, since iirc a user specifically asked for the strictest mode, so it'd probably be worth doing a poll in the community before making a decision here.

**Comment by lydia-duncan** [comment]
@bradcray - should we try to get this discussed before 2.0?

**Comment by mppf** [comment]
> Related: The `prototype` keyword has always been a slightly odd one, and @mppf proposes that perhaps it should be an annotation. I'm intrigued by that...

Now this has its own issue: #23026.

The 3 modes are:
 * the mode you get with `prototype module` (I'd call this prototype mode or something; the docs call it fatal mode). I think we want to continue to have this mode available in some way.
 * the mode you get with a regular `module` (I'd call this the default mode, today; the docs call it relaxed mode).  I think we want to continue to have this mode available in some way.
 * a very strict mode that you can only get with `pragma "error mode strict"`. It is documented in https://chapel-lang.org/docs/technotes/errorHandling.html#strict-mode . It's unclear to me if we need to deprecate it or if we can simply remove it.

**Comment by bradcray** [comment]
I tend to agree that strict mode could be deprecated or retired.  I don't know that it needs much discussion, in the sense that I'd be surprised if hardly anyone on the team knew/remembered that we had it.  I've been very happy with the two modes we have and can't imagine wanting another (though I also can't remember what the third is and don't feel particularly motivated to go remind myself a.t.m.).

Since Michael's reminded us here that it's enabled with a pragma and pragmas themselves are unstable, I think we could remove the tag here and call it done for the fall if we didn't want to go to the trouble to rip it out a.t.m.  It might also be prudent to put a mention of its instability in the docs, or just to remove the docs about it altogether and make it an "undocumented feature" until we do bother removing it.

**Comment by lydia-duncan** [comment]
A user did ask us specifically for the strictest mode, which is my main hesitation

**Comment by bradcray** [comment]
Are you hesitating about the idea of removing it, or nervous about it remaining in an unstable state for the fall release?

**Comment by lydia-duncan** [comment]
Hesitating about removing it

**Comment by bradcray** [comment]
I still think we could remove the 2.0 label from this on the argument that it's already marked as unstable (all pragmas are unstable) and it seems unlikely we'll have the cycles to do much more here.  Unless someone wants to mark it as unstable in the technote.
